### PR TITLE
Add new module for resources

### PR DESCRIPTION
**Pull request by abhishek1508**
## Description

Add a new module that would contain all the resource files. (https://github.com/mapbox/navigation-sdks/issues/197)

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Add a new module that would contain all the resource files.

### Implementation

Add a new module that would contain all the resource files.

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-resource/build.gradle [hunk 1]
```diff
@@ -0,0 +1,26 @@
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+android {
+    compileSdkVersion 29
```
Comment (Guardiola31337) [inline libnavigation-resource/build.gradle]: We should use `ext` values defined in `[dependencies.gradle](https://github.com/mapbox/mapbox-navigation-android/blob/master/gradle/dependencies.gradle). Same thing applies to `dependenciesList` 👇

File: libnavigation-resource/build.gradle [hunk 2]
```diff
@@ -0,0 +1,26 @@
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
```
Comment (Guardiola31337) [inline libnavigation-resource/build.gradle]: Do we need this `plugin`?

File: libnavigation-resource/src/main/res/drawable/ic_maneuver_merge_left.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    android:name="merge"
+    android:width="50dp"
+    android:height="50dp"
+    android:viewportHeight="50"
+    android:viewportWidth="50">
+    <group
+        android:name="group3"
+        android:translateX="25.07"
+        android:translateY="24.48">
+        <!--<path-->
```
Comment (Guardiola31337) [inline libnavigation-resource/src/main/res/drawable/ic_maneuver_merge_left.xml]: Remove commented out code?

File: libnavigation-resource/src/main/res/drawable/ic_maneuver_merge_right.xml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    android:name="merge"
+    android:width="50dp"
+    android:height="50dp"
+    android:viewportHeight="50"
+    android:viewportWidth="50">
+    <group
+        android:name="group3"
+        android:translateX="24.93"
+        android:translateY="24.48">
+        <!--<path-->
```
Comment (Guardiola31337) [inline libnavigation-resource/src/main/res/drawable/ic_maneuver_merge_right.xml]: Remove commented out code?

File: libnavigation-resource/src/main/res/drawable/ic_maneuver_merge_right.xml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        <!--android:name="bezier"-->
```
Comment (kmadsen) [inline libnavigation-resource/src/main/res/drawable/ic_maneuver_merge_right.xml]: do we need this?

**Pull request merged**